app/models/ingest_job.rb

```diff
@@ -417,81 +438,97 @@ def set_study_initialized
+      metadata_identifier = "#{study_file.bucket_location}:#{study_file.id}"
+      study.study_files.where(file_type: 'Cluster', parse_status: 'parsed').each do |cluster_file|
+        cluster = ClusterGroup.find_by(study_id: study.id, study_file_id: cluster_file.id)
         if cluster.can_subsample? && !cluster.is_subsampling?
           # set cluster.is_subsampling = true to avoid future race conditions
           cluster.update(is_subsampling: true)
           file_identifier = "#{cluster_file.bucket_location}:#{cluster_file.id}"
-          Rails.logger.info "Launching subsampling ingest run for #{file_identifier} after #{self.action} of #{metadata_identifier}"
-          submission = ApplicationController.papi_client.run_pipeline(study_file: cluster_file, user: self.user,
+          Rails.logger.info "Launching subsampling ingest run for #{file_identifier} after #{action} of #{metadata_identifier}"
+          submission = ApplicationController.papi_client.run_pipeline(study_file: cluster_file, user: user,
                                                                       action: :ingest_subsample)
           Rails.logger.info "Subsampling run initiated: #{submission.name}, queueing Ingest poller"
-          IngestJob.new(pipeline_name: submission.name, study: self.study, study_file: cluster_file,
-                        user: self.user, action: :ingest_subsample, reparse: self.reparse,
-                        persist_on_fail: self.persist_on_fail).poll_for_completion
+          IngestJob.new(pipeline_name: submission.name, study: study, study_file: cluster_file,
+                        user: user, action: :ingest_subsample, reparse: reparse,
+                        persist_on_fail: persist_on_fail).poll_for_completion
         end
       end
     end
   end
 
   # Set correct subsampling flags on a cluster after job completion
   def set_subsampling_flags
-    case self.action
-    when :ingest_subsample
-      cluster_group = ClusterGroup.find_by(study_id: self.study.id, study_file_id: self.study_file.id)
-      if cluster_group.is_subsampling? && cluster_group.find_subsampled_data_arrays.any?
-        Rails.logger.info "Setting subsampled flags for #{self.study_file.upload_file_name}:#{self.study_file.id} (#{cluster_group.name}) for visualization"
-        cluster_group.update(subsampled: true, is_subsampling: false)
+    cluster_group = ClusterGroup.find_by(study_id: study.id, study_file_id: study_file.id)
+    if cluster_group.is_subsampling? && cluster_group.find_subsampled_data_arrays.any?
+      Rails.logger.info "Setting subsampled flags for #{study_file.upload_file_name}:#{study_file.id} (#{cluster_group.name}) for visualization"
+      cluster_group.update(subsampled: true, is_subsampling: false)
+    end
+  end
+
+  # set corresponding differential expression flags on associated annotation
+  def set_differential_expression_flags
```

Comment:
this might also belong in the new differential_expression_service.rb file

Reply:
I disagree, since this is setting state at the end of an ingest job.  All other such methods live in `IngestJob`.